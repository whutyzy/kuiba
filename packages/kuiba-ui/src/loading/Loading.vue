<template>
    <div :class="['kuiba-loading', `kuiba-loading--${type}`]">
        <span :class="['kuiba-loading__spinner', `kuiba-loading__spinner--${type}`]" :style="spinnerStyle">
            <svg v-if="type === 'circle'" :class="`kuiba-loading__${type}`" viewBox="25 25 50 50">
                <circle cx="50" cy="50" r="20" fill="none"></circle>
            </svg>
            <template v-else>
                <i v-for="i in 12" :key="i" :class="['kuiba-loading__line', `kuiba-loading__line--${i}`]"></i>
            </template>
        </span>
    </div>
</template>

<script lang="ts">
import { defineComponent, PropType, computed } from 'vue'
import { getSizeStyle } from '../utils/format/unit'
export default defineComponent({
    name: 'KuibaLoading',
    props: {
        type: {
            type: String as PropType<'circle' | 'spinner'>,
            default: 'circle'
        },
        size: {
            type: [Number, String]
        },
        color: {
            type: String,
            default: 'currentColor'
        }
    },
    setup(props) {
        const spinnerStyle = computed(() => {
            return Object.assign({ color: props.color }, getSizeStyle(props.size))
        })

        return { spinnerStyle }
    }
})
</script>

<style lang="scss">
@import './loading.scss';
</style>
