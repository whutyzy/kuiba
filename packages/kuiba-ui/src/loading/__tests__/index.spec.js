import example from '../example'
import Loading from '../index'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'

test('test loading example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test loading plugin', () => {
  const app = createApp({}).use(Loading)
  expect(app.component(Loading.name)).toBeTruthy()
})
