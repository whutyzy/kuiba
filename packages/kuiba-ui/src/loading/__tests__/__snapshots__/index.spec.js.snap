// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle-props 1`] = `"<div class=\\"kuiba-loading kuiba-loading--circle\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--circle\\" style=\\"color: rgb(239, 239, 239); width: 24px; height: 24px;\\"><svg class=\\"kuiba-loading__circle\\" viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>"`;

exports[`spinner-props 1`] = `"<div class=\\"kuiba-loading kuiba-loading--spinner\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--spinner\\" style=\\"color: rgb(239, 239, 239); width: 24px; height: 24px;\\"><i class=\\"kuiba-loading__line kuiba-loading__line--1\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--2\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--3\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--4\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--5\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--6\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--7\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--8\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--9\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--10\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--11\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--12\\"></i></span></div>"`;

exports[`test loading example 1`] = `
"<div class=\\"kuiba-demo-block\\">
  <h4 class=\\"kuiba-demo-block__title\\">类型</h4>
  <div class=\\"kuiba-demo-block__container\\">
    <div class=\\"kuiba-loading kuiba-loading--circle\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--circle\\" style=\\"color: currentColor;\\"><svg class=\\"kuiba-loading__circle\\" viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
    <div class=\\"kuiba-loading kuiba-loading--spinner\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--spinner\\" style=\\"color: currentColor;\\"><i class=\\"kuiba-loading__line kuiba-loading__line--1\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--2\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--3\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--4\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--5\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--6\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--7\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--8\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--9\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--10\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--11\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--12\\"></i></span></div>
  </div>
</div>
<div class=\\"kuiba-demo-block\\">
  <h4 class=\\"kuiba-demo-block__title\\">颜色</h4>
  <div class=\\"kuiba-demo-block__container\\">
    <div class=\\"kuiba-loading kuiba-loading--circle\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--circle\\" style=\\"color: orange;\\"><svg class=\\"kuiba-loading__circle\\" viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
    <div class=\\"kuiba-loading kuiba-loading--spinner\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--spinner\\" style=\\"color: pink;\\"><i class=\\"kuiba-loading__line kuiba-loading__line--1\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--2\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--3\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--4\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--5\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--6\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--7\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--8\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--9\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--10\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--11\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--12\\"></i></span></div>
  </div>
</div>
<div class=\\"kuiba-demo-block\\">
  <h4 class=\\"kuiba-demo-block__title\\">大小</h4>
  <div class=\\"kuiba-demo-block__container\\">
    <div class=\\"kuiba-loading kuiba-loading--circle\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--circle\\" style=\\"color: currentColor; width: 48px; height: 48px;\\"><svg class=\\"kuiba-loading__circle\\" viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
    <div class=\\"kuiba-loading kuiba-loading--spinner\\"><span class=\\"kuiba-loading__spinner kuiba-loading__spinner--spinner\\" style=\\"color: currentColor; width: 48px; height: 48px;\\"><i class=\\"kuiba-loading__line kuiba-loading__line--1\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--2\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--3\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--4\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--5\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--6\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--7\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--8\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--9\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--10\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--11\\"></i><i class=\\"kuiba-loading__line kuiba-loading__line--12\\"></i></span></div>
  </div>
</div>"
`;
